commerce_coordinator/apps/titan/tasks.py

```diff
@@ -75,8 +83,21 @@ def payment_processed_save_task(payment_number, payment_state, response_code):
 
     titan_api_client = TitanAPIClient()
 
-    titan_api_client.update_payment(
-        payment_number=payment_number,
-        payment_state=payment_state,
-        response_code=response_code
-    )
+    try:
+        payment = titan_api_client.update_payment(
+            payment_number=payment_number,
+            payment_state=payment_state,
+            response_code=response_code
+        )
+        # Set cache after successfully updating payment state in Titan's system.
+        payment_state = payment['state']
+        if payment_state == PaymentState.COMPLETED.value:
+            payment_state_paid_cache_key = get_paid_payment_state_cache_key(payment_number)
+            TieredCache.set_all_tiers(payment_state_paid_cache_key, payment, settings.DEFAULT_TIMEOUT)
+        elif payment_state == PaymentState.FAILED.value:
+            payment_state_processing_cache_key = get_processing_payment_state_cache_key(payment_number)
+            TieredCache.set_all_tiers(payment_state_processing_cache_key, payment, settings.DEFAULT_TIMEOUT)
```

Comment:
after updating the payment in Titan, We are setting the cache.
@pshiu Thank you for reminding me.